### PR TITLE
fix(VField): correct height for compact/comfortable densities

### LINKED ISSUE #17604: [Bug Report][3.3.4] VSelect Value Vertical Alignment
### Environment
**Vuetify Version:** 3.3.4
**Vue Version:** 3.3.4
**Browsers:** Firefox 113.0
**OS:** Mac OS 10.15

### Steps to reproduce
The value and placeholder in a VSelect is no longer vertically aligned, but it up a few pixels. 
It functions fine for VTextField and the VAutocomplete

### Expected Behavior
The text should be in the center

### Actual Behavior
The text is top aligned

### Reproduction Link
[https://play.vuetifyjs.com/#...](https://play.vuetifyjs.com/#eNqtkk1u2zAQha8y5cYtYIlF024Mp0h3vUEXYRaMOK6Z8A/kSIlh+O4ZSjYiOF4ESFYS3zy+j5zh7V78SakdehQrsSb0yWnC3yoArIdGpzT+josuBtI2YIZCO4fXSjxZQ9sV/Pj1PT0rcXSOXsJnajYWnQEO7HAbncHMW/6ic1EJGHS2OhArsSfHqYZFg6FY2rHYRZ90R6zJeWxBhx19aqTuKdaSQ8KTDJcJr+VLqNfqBeapeEKv5ayfU7dZmdq9lrMx8LJ02SaCgtRP07A+xUywh4wbOMAmRw8LnuCi2gE4txD48h+uq+PrYjw+/IvZmS+LbxUwRXKYWIoprfE6tQ8lBn4G+5qijoWixApGpWpMqWsltkSprKTsTOBtBp0dchuQZEhe3rBN5j6Q9diY6G+u2qv2pzS20FxusfjmPsengplDlFjOMGQ3u3egjs454Sg1Tt+XingTLZk3YG4yBp4v5vfe6GzbnHlWenOzSj+ocOB+83N5LGet7srY5tvTIT905TGsJt1V4GEpRiOTp+/dC32TUWY=)

### LINKED ISSUE #17837: [Bug Report][3.3.9] text/placeholder in v-textarea and v-text-field are not aligned
### Environment
**Vuetify Version:** 3.3.9
**Last working version:** 3.3.3
**Vue Version:** 3.3.4
**Browsers:** Chrome 114.0.0.0
**OS:** Windows 10

### Steps to reproduce
load reproduction playground

### Expected Behavior
text/placeholder in input and textarea should be aligned

### Actual Behavior
text/placeholder in input and textarea are not aligned

### Reproduction Link
[https://play.vuetifyjs.com/#](https://play.vuetifyjs.com/?preview=false#eNrtVUtu2zAQvQqrLtwCppQgbYGqTpH2Cl10EWVBiyOLjfgBOVJiBL57h5JT0wra1Mg2Cxvi/N7M4yN5/ZB9cy4fesjKbIWgXScQvlaGsdXAhXPj57iorUGhDPi9aTR6e/dnuY/qGP3CZZV9qrLEN3oR7lF4EKmZsYFrK6HL0StNecq4HqvsOKYkpFj0fO6ghmtobSfBk/unVwhsa3vPNIQgNpDn+TxlEF4JgxRue+xoJDmPkGCCwi1F1FY31qNYdzAPEj1avqG+js2tksAlEFddSD1HXBAbj2Szt86DAyO5Mim7CW2KuGfxjxrSUnGtam9daw31FNm2cfSNhy0tizlOkezqzPWGc/aDkNl3NIzzv3dIQjipwUDRJ7aWmg9CScVFZip4qtx4o6CTr4J7Fdyzgpuk8g/JxXV6462kGg7RdSdCVMwdPz87S/cu4LYDcqSYLahNiyU7d/dfUruzpANlTckadQ/yyIfWlezjPKODhuqcHdnWor4lofRGloy4AJN4D3080jONmd7u0TJd/rPdXIXaK4csAPbT26C0I7myB+ahWbI7gXXLdqzxVrMFPSuLSQNUPiAbDxq7jKHvFov3sfpUjyply2wqxbVw+a9gDb1IDzGXDujoCFVWstESbVQ7rqusRXShLIpaGkqjY60GnxvAwjhdXFFY4XuDStMZsfrqIr/IPxRSBUzNOQTN1/HEg6ciVbZMYFA1JKZnofaRI8LnCWFv4p1YhwjxpHRBeAN47km+4MH/70SztHSqmevJZBF9V5kd8U030W2YUV2HkebrxyZfNPJYLFa6iYC7ZTb2SchjQnbzG6B7fYI=)

### LINKED ISSUE #17898: [Bug Report][3.3.10] fields are still broken
### Environment
**Vuetify Version:** 3.3.10
**Last working version:** 3.3.10
**Vue Version:** 3.2.47
**Browsers:** Chrome 114.0.0.0
**OS:** Mac OS 10.15.7

### Steps to reproduce
use fields components as usual with vuetify@3.3.3
update to vuetify@3.3.10

### Expected Behavior
Field components heights and composition remains the same across minor version change

### Actual Behavior
Field components heights was unexpectedly changed and component inner value placement is broken at least between VTextField and VSelect

### Reproduction Link
[https://codepen.io/yurks/pen/xxQQgeY](https://codepen.io/yurks/pen/xxQQgeY)

### Other comments

A number of closed issues attached to #17412, but fields styling is completely broken in comparing to v3.3.3

**3.3.3 -> 3.3.5**
image

As you can see with reproduction Link:
- heights are different across different field variants
- heights are different across different field densities 
- heights are different with all versions up to v3.3.3
- VSelect value placement has incorrect position
- default density height was reduced to correct 56px according #17760. But only for default variant and density

### PR DESCRIPTION
**Pull request by yuwu9145**
fixes #17898
fixes #17837
keep fix #17604

### Changes:

- This is a big refactoring aims to preserve all previous fixes while addressing new density-related height issues.
- underlined/plain variants default height as 48px, and smaller than other variants in each density cohort
- all other variants have heights as 56(default)/48(comfortable)/40(compact)
- Use CSS gap to create spacing between rows when multiple chips are selected and they stack on top of each other. It allows previous logic of aligning input with chips to be removed.

### Regression checklist:

- https://github.com/vuetifyjs/vuetify/pull/17175#issuecomment-1551147290 - ✅ 
- #17892 - ✅
- [percy change](https://percy.io/c458cb73/vuetify/builds/30072079/changed/1664810483?browser=chrome&browser_ids=48&subcategories=unreviewed%2Cchanges_requested&viewLayout=overlay&viewMode=new&width=1280&widths=375%2C1280)
- local doc testing ✅ 

## Markup:

```vue
<template>
  <v-app>
    <v-container>
      <h1>With Labels</h1>
      <h3>Density: default</h3>
      <v-row>
        <v-col>
          <v-select
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            label="Default"
            variant="outlined"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            label="Default"
            placeholder="Hello"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            label="Default"
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            label="Default"
            variant="solo"
          />
        </v-col>
      </v-row>
      <h3>Density: comfortable</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="outlined"
            density="comfortable"
            label="Comfortable"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="comfortable"
            label="Comfortable"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="comfortable"
            variant="underlined"
            label="Comfortable"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="comfortable"
            variant="solo"
            label="Comfortable"
          />
        </v-col>
      </v-row>
      <h3>Density: compact</h3>
      <v-row>
        <v-col>
          <v-select
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            variant="outlined"
            label="Compact"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            label="Compact"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="compact"
            label="Compact"
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="compact"
            label="Compact"
            variant="solo"
          />
        </v-col>
      </v-row>
      <h1>Without Labels</h1>
      <h3>Density: default</h3>
      <v-row>
        <v-col>
          <v-select
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            variant="outlined"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="solo"
          />
        </v-col>
      </v-row>
      <h3>Density: comfortable</h3>
      <v-row>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            variant="outlined"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="comfortable"
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="comfortable"
            variant="solo"
          />
        </v-col>
      </v-row>
      <h3>Density: compact</h3>
      <v-row>
        <v-col>
          <v-select
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            variant="outlined"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="compact"
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-text-field
            :model-value="'California'"
            placeholder="Hello"
            density="compact"
            variant="solo"
          />
        </v-col>
      </v-row>
      <h1>Autocomplete (With Labels)</h1>
      <h3>Density: default</h3>
      <v-row>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            label="Default"
            variant="outlined"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            label="Default"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            placeholder="Hello"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            label="Default"
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            label="Default"
            variant="solo"
          />
        </v-col>
      </v-row>
      <h3>Density: comfortable</h3>
      <v-row>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            variant="outlined"
            chips
            multiple
            density="comfortable"
            label="Comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            chips
            multiple
            density="comfortable"
            label="Comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            density="comfortable"
            variant="underlined"
            chips
            multiple
            label="Comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            density="comfortable"
            chips
            multiple
            variant="solo"
            label="Comfortable"
          />
        </v-col>
      </v-row>
      <h3>Density: compact</h3>
      <v-row>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            variant="outlined"
            chips
            multiple
            label="Compact"
            density="compact"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            label="Compact"
            chips
            multiple
            density="compact"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            density="compact"
            chips
            multiple
            label="Compact"
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            density="compact"
            label="Compact"
            chips
            multiple
            variant="solo"
          />
        </v-col>
      </v-row>
      <h1>Autocomplete (Without Labels)</h1>
      <h3>Density: default</h3>
      <v-row>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            variant="outlined"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            placeholder="Hello"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            variant="solo"
          />
        </v-col>
      </v-row>
      <h3>Density: comfortable</h3>
      <v-row>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            variant="outlined"
            chips
            multiple
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            placeholder="Hello"
            chips
            multiple
            density="comfortable"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            density="comfortable"
            variant="underlined"
            chips
            multiple
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            density="comfortable"
            chips
            multiple
            variant="solo"
          />
        </v-col>
      </v-row>
      <h3>Density: compact</h3>
      <v-row>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            variant="outlined"
            chips
            multiple
            density="compact"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            chips
            multiple
            density="compact"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            density="compact"
            chips
            multiple
            variant="underlined"
          />
        </v-col>
        <v-col>
          <v-autocomplete
            :model-value="'California'"
            placeholder="Hello"
            :items="['California', 'Colorado', 'Florida', 'Georgia', 'Texas', 'Wyoming']"
            density="compact"
            chips
            multiple
            variant="solo"
          />
        </v-col>
      </v-row>
    </v-container>
  </v-app>
</template>
```

### CODE REVIEW

**Comment by yuwu9145** [comment]
I will mark this PR ready after #17892 is in and no regression is inspected

**Comment by yuwu9145** [comment]
regression checklist:

- https://github.com/vuetifyjs/vuetify/pull/17175#issuecomment-1551147290
- #17412
- https://github.com/vuetifyjs/vuetify/pull/17607#issuecomment-1596101048

File: packages/vuetify/src/components/VTextarea/__tests__/VTextarea.spec.cy.tsx [hunk 1]
```diff
@@ -27,8 +27,8 @@ describe('VTextarea', () => {
       .should(el => expect(el.outerHeight()).to.equal(56))
       .click()
       .type('Lorem ipsum dolor sit amet consectetur adipisicing elit. ')
-      .should(el => expect(el.outerHeight()).to.equal(78))
+      .should(el => expect(el.outerHeight()).to.equal(80))
```
Comment (yuwu9145) [inline packages/vuetify/src/components/VTextarea/__tests__/VTextarea.spec.cy.tsx:30]: Input padding top/bottom has been adjusted to 16px by default, so:

- For one row, height (56px) = line-height (24px) + padding-top (16px) + padding-bottom (16px)
- For two rows,  height (80px) = line-height (24px) * 2 + padding-top (16px) + padding-bottom (16px)

**Comment by johnleider** [comment]
The only issue I see is that filled variant label moved up too far:

This PR:
Screenshot_18

v3.3.15:
Screenshot_19

**Review by johnleider — APPROVED** [review]

File: packages/vuetify/src/components/VField/VField.sass [hunk 2]
```diff
@@ -82,7 +82,7 @@
     @at-root
       @include tools.density('v-input', $input-density) using ($modifier)
         @at-root #{selector.nest(&, $root)}
-          --v-field-padding-bottom: #{15px + $modifier * .5}
+          --v-field-padding-bottom: #{16px + $modifier * .5}
```
Comment (KaelWD) [inline packages/vuetify/src/components/VField/VField.sass:85]: Default density seems slightly off center, should this still be 15?

Comment (yuwu9145) [reply packages/vuetify/src/components/VField/VField.sass:85]: Before the input was assigned uneven chips margin top/bottom, in order to be aligned with chips, so it looks bit off center.

Now css gap is being used and chips margin top/bottom are completely removed.

Input/chips should be always aligned center in outlinded variant or without labels.

**Review by johnleider — APPROVED** [review]

File: packages/vuetify/src/components/VTextField/VTextField.sass [hunk 3]
```diff
@@ -86,7 +73,7 @@
 /* region MODIFIERS */
 .v-text-field
   &--plain-underlined
-    --v-field-padding-top--plain-underlined: 6px
+    --v-field-padding-top--plain-underlined: 0px
```
Comment (KaelWD) [inline packages/vuetify/src/components/VTextField/VTextField.sass:76]: Wait why is this even here? All components using VField have `.v-text-field--plain-underlined` set, so the default `--v-field-padding-top: 6px` in VField.sass will only be hit if using VField standalone. 
Also I missed it in #14664 but this whole section should really be moved to VInput, currently it'll only work if VTextField is used (which is likely but not 100%)

**Comment by yuwu9145** [comment]
I did another round of testing, and the new change did not break anything, which is good.

However, looking ahead in this task, since any new change requires a comprehensive and time-consuming testing process, I think priority should be addressing UI issues and generating a reliable Percy base report on the master branch, as opposed to resolving tech debt. Subsequent improvements can be made on top of the solid Percy snapshots.

This approach will help us avoid repeated testing.

**Comment by KaelWD** [comment]
Yeah I couldn't find any more obvious problems, should we just get this in then?

**Comment by yuwu9145** [comment]
I tested again, it's in good shape, feel free to get it in at appropriate time

**Review by johnleider — APPROVED** [review]

**Pull request merged**